File: 10/06/10/06/10-06-2025_Maximum_Difference_Between_Even_and_Odd_Frequency_I.cpp
class Solution {
public:
    int maxDifference(string s) {
        unordered_map<char, int> m;
        for(int i=0; i<s.length(); i++){
            m[s[i]]++;
        }
        int maxf=0;
        int minf=s.length();
        for(auto &d:m){
            if((d.second & 1) ==1) maxf=max(maxf, d.second);
            else{
                minf= min(minf, d.second);
            }
        }
        if(minf==s.length()) return maxf;
        return maxf-minf;
        
    }
};

File: 10/06/10/06/10-06-2025_Maximum_Difference_Between_Even_and_Odd_Frequency_I.java
import java.util.HashMap;

class Solution {
    public int maxDifference(String s) {
        HashMap<Character, Integer> freq = new HashMap<>();
        
        for (char c : s.toCharArray()) {
            freq.put(c, freq.getOrDefault(c, 0) + 1);
        }
        
        int maxf = 0;
        int minf = s.length();
        
        for (int count : freq.values()) {
            if ((count & 1) == 1) {
                maxf = Math.max(maxf, count);
            } else {
                minf = Math.min(minf, count);
            }
        }
        
        if (minf == s.length()) return maxf;
        return maxf - minf;
    }
            }
